pkg/handlers/v1/attribute.go

```diff
@@ -0,0 +1,40 @@
+package v1
+
+import (
+	"context"
+
+	"github.com/asecurityteam/nexpose-asset-attributor/pkg/assetattributor"
+	"github.com/asecurityteam/nexpose-asset-attributor/pkg/domain"
+	"github.com/asecurityteam/nexpose-asset-attributor/pkg/logs"
+)
+
+// AttributeHandler handles the Attribution endpoint for nexpose-asset-attributor
+type AttributeHandler struct {
+	AssetAttributor domain.AssetAttributor
+	LogFn           domain.LogFn
+	StatFn          domain.StatFn
+}
+
+// Handle processes the incoming domain.NexposeAssetVulnerabilities instance, queries available asset inventory systems via its
+// AssetAttributor, and returns a domain.AttributedAssetVulnerabilities instance, annotated with the business context of the asset
+// at Nexpose scan time.
+func (h *AttributeHandler) Handle(ctx context.Context, assetVulns domain.NexposeAssetVulnerabilities) (domain.NexposeAttributedAssetVulnerabilities, error) {
+	logger := h.LogFn(ctx)
+	stater := h.StatFn(ctx)
+
+	attributedAssetVulns, err := h.AssetAttributor.Attribute(ctx, assetVulns)
+	if err != nil {
+		if notFoundError, ok := err.(*assetattributor.AssetNotFoundError); ok {
+			logger.Error(logs.AssetNotFoundError{Reason: notFoundError.Error()})
+			stater.Count("event.nexposeassetattributor.attribution_failure.asset_not_found", 1)
```

Comment:
Why does this need a custom metric? Are we not able to rely on the metrics provided by the proxy/gateway?

Reply:
Maybe we can, but I don't know what metrics are provided by the gateway. Is there a single place I can look for a full listing?

Reply:
Lambda will record an error in cloudwatch for every non-nil error value you return if running in AWS Lambda. Likewise, the serverfull-gateway will record the status code which you will have to define based on the errors returned (or a general case of 5xx) if running with serverfull. My encouragement is that we don't add new metrics where existing, standard metrics can suffice. If the standard metrics are insufficient then, by all means, add a metric. I automatically question all custom metrics to make sure the intent isn't already covered.

To directly answer your question, the full list of metrics for serverfull-gateway is here: https://github.com/asecurityteam/transportd#route-settings which, internally, is using https://github.com/asecurityteam/httpstats#http-client-1 if you want a deeper dive. For AWS Lambda, the full list of metrics is here: https://docs.aws.amazon.com/lambda/latest/dg/monitoring-functions-metrics.html